controllers/tempest_controller.go

```diff
@@ -515,6 +512,34 @@ func mergeWithWorkflow[T any](value T, workflowValue *T) T {
 	return *workflowValue
 }
 
+func getTempestRun(instance *testv1beta1.Tempest, workflowStepNum int) testv1beta1.TempestRunSpec {
+	if workflowStepNum < len(instance.Spec.Workflow) {
+		wtRun := instance.Spec.Workflow[workflowStepNum].TempestRun
+		tRun := changeType(wtRun)
+		return tRun
+	}
+	return instance.Spec.TempestRun
+}
+
+func changeType(wtRun testv1beta1.WorkflowTempestRunSpec) testv1beta1.TempestRunSpec {
```

Comment:
I think we should aim at making this function more generic by replacing the types with `interface{}`. The caller of the function will be responsible of casting the result to a correct type.

```suggestion
func changeType(instance interface{}) instance interface{} {
```